yb-voyager/src/query/queryparser/ddl_processor.go

```diff
@@ -849,16 +851,52 @@ func (v *ViewProcessor) Process(parseTree *pg_query.ParseResult) (DDLObject, err
 	if !ok {
 		return nil, fmt.Errorf("not a CREATE VIEW statement")
 	}
+
+	viewSchemaName := viewNode.ViewStmt.View.Schemaname
+	viewName := viewNode.ViewStmt.View.Relname
+	qualifiedViewName := lo.Ternary(viewSchemaName == "", viewName, viewSchemaName+"."+viewName)
+	/*
+		view_stmt:{view:{schemaname:"public" relname:"invoker_view" inh:true relpersistence:"p" location:12}
+		query:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"id"}} location:95}} location:95}}
+		from_clause:{...}
+		where_clause:{...}
+		options:{def_elem:{defname:"security_invoker" arg:{string:{sval:"true"}} defaction:DEFELEM_UNSPEC location:32}}
+		options:{def_elem:{defname:"security_barrier" arg:{string:{sval:"false"}} defaction:DEFELEM_UNSPEC location:57}}
+		with_check_option:NO_CHECK_OPTION}
+	*/
+
+	log.Infof("checking the view '%s' is security invoker view", qualifiedViewName)
```

Comment:
As per my understanding, the benefit of proto detection comes from writing generic code :
- without worrying about what type of statement it is. For example, be it a view/mview/index/table, you can get the defName of the defElemNodes.
- without worrying about which fields do I have to look at to get the "with options". In case of a view, it seems like `ViewStmt.Options`. (In some others that I've checked it also seems to be the `Options` field)

Therefore, I see the following approaches:
1. Generic proto detection approach for defElemNodes: How about writing a generic proto detector of defElem nodes? DefElem seems like the generic option definition node. The current logic what you've written can be re-used for other statement types as well in future. (for example, to detect a particular option being unsupported in CREATE TABLE).
 https://github.com/postgres/postgres/blob/578a7fe7b6f8484f6d7caa2fda288abb3fe87aa0/src/include/nodes/parsenodes.h#L795 
 

3. Write a generic proto ViewStmtNodeDetector. All issues pertaining to views can live there. This would be in parallel to the ViewProcessor that we already have today.

4. In the current ddl processor, use the viewStmt node directly. I personally see this also as a viable option because this is exactly what we would have had to do if we wrote a ViewStmtNodeDetector (option 2 above);
```
	securityInvokerView := false
	for _, node := range viewNode.ViewStmt.Options {
		defElemNode := node.GetDefElem()
		if defElemNode != nil {
			if defElemNode.GetDefname() == "security_invoker" {
				securityInvokerView = true
				break
			}
		}
	}
```


I guess my main point is that we aren't gaining much by writing a view-specific proto detection logic. Either go with option 1(write generic proto detection logic), or option 2/3 (view-specific logic on ViewStmt node.)

Reply:
Couple of points:
1. I think moving forward we wanted to have mix of both(protomsg and struct approach), where something is directly available in the struct use it, and if it requires some lookup or wherever generic code is required then use protomsg approach.
So here going with the option 2/3 does not to me aligning with that, it again pulls us to forcing struct approach.

2. I didn't get exactly why do we need to introduce something like `ViewStmtNodeDetector`? What i understood from implementation in `detectors_ddl.go` that we will specific detectors for each object type like we have `ViewIssueDetector`, and enhance that to detect all issues related to views.

3. Regarding - `Generic proto detection approach for defElemNodes` , i did think for adding this piece of code so that its reusable, since the `DefElem` node can be present at other places like `CREATE TABLE ... WITH (fillfactor=70)...` but couldn't come up how/where in existing code.
Seems like you are suggesting it placing parallel to `FuncCallDetector`, `ColumnDef` detectors. 
The problem is that the `security invoker` is just a parameter, i am not sure if it can be present with other nodes also, or just specific to VIEWs. (For example: [functions definition have security invoker](https://www.postgresql.org/docs/current/sql-createfunction.html#:~:text=INPUT%20%7C%20STRICT%20%7D%0A%20%20%20%20%7C%20%7B%20%5B%20EXTERNAL%20%5D-,SECURITY%20INVOKER,-%7C%20%5B%20EXTERNAL%20%5D%20SECURITY%20DEFINER))

Reply:
going with `Generic proto detection approach for defElemNodes` as discussed